### PR TITLE
Add Column.is_number/is_float (#1969)

### LINKED ISSUE #1969: Column.is_integer() method
The [Column](https://docs.getdbt.com/docs/api-variable#section-instance-methods-) class has an instance method, `is_numeric()` that returns true if the column is a fixed-precision Numeric type (eg. numeric), else False (including integer and float types).

There are instances where I want to determine:
- Is this column generally numeric? I.e. one of: decimal, float, integer
- Is this column an integer? i.e. one of bigint, smallint, integer (e.g. on Redshift).

The second one is easy: let's add an `is_integer()` method.
But I don't know how to name the first method, since `is_numeric()` is already reserved for numeric (i.e. decimal) columns. Can we rename `is_numeric` to `is_decimal`, or will that mess up too many things?

### PR DESCRIPTION
**Pull request by beckjake**
Fixes #1969 

A fun question to think about: on snowflake, is nothing an integer, or is everything? Snowflake doesn't technically have non-`NUMBER` integer types: https://docs.snowflake.net/manuals/sql-reference/data-types-numeric.html#int-integer-bigint-smallint-tinyint-byteint - On the database end, `information_schema.columns` reports all these as being DECIMAL(38, 0). I've gone with: everything on snowflake is `numeric`, `float`, or `text` and none of it is `integer`.

### CODE REVIEW

File: core/dbt/adapters/base/column.py [hunk 1]
```diff
@@ -52,8 +50,27 @@ def is_string(self) -> bool:
...
+            'smallint', 'integer', 'bigint',
+            'smallserial', 'serial', 'bigserial',
+            # aliases
+            'int2', 'int4', 'int8',
+            'serial2', 'serial4', 'serial8',
+        ]
+
     def is_numeric(self) -> bool:
-        return self.dtype.lower() in ['numeric', 'number']
+        return self.dtype.lower() in ['numeric', 'decimal']
```
Comment (drewbanin) [inline core/dbt/adapters/base/column.py]: This change is the only thing in this PR that I'm not sure about -- do you know why `number` was in here before? I tried looking through the blame but couldn't find the origin.... AFAICT `number` is not a valid type on pg/redshift/snowflake/bq.....?

Comment (beckjake) [reply core/dbt/adapters/base/column.py]: It's only valid on snowflake, and previously this method applied for Snowflake as well as postgres/redshift. That's why I removed it when I split snowflake out.

File: plugins/bigquery/dbt/adapters/bigquery/column.py [hunk 2]
```diff
@@ -99,8 +99,15 @@ def data_type(self) -> str:
     def is_string(self) -> bool:
         return self.dtype.lower() == 'string'
 
+    def is_integer(self) -> bool:
+        # snowflake technicality: These are all synonyms with NUMBER(38, 0)
```
[the remark this answers is not in the document]
Comment (beckjake) [reply plugins/bigquery/dbt/adapters/bigquery/column.py]: Nope!

[the remark this answers is not in the document]
Comment (beckjake) [reply plugins/bigquery/dbt/adapters/bigquery/column.py]: Nope!

**Pull request merged**